python/cuml/manifold/t_sne.pyx

```diff
@@ -436,24 +429,25 @@ class TSNE(Base):
                 Embedding of the training data in low-dimensional space.
         """
         self.fit(X, convert_dtype=convert_dtype)
+        out_type = self._get_output_type(X)
 
         if isinstance(X, cudf.DataFrame):
-            if isinstance(self.embedding_, cudf.DataFrame):
-                return self.embedding_
+            if isinstance(self._embedding_, cudf.DataFrame):
+                return self._embedding_
             else:
-                return cudf.DataFrame.from_gpu_matrix(self.embedding_)
+                return cudf.DataFrame.from_gpu_matrix(self._embedding_)
         elif isinstance(X, np.ndarray):
-            data = self.embedding_.copy_to_host()
-            del self.embedding_
+            data = self._embedding_.to_output(out_type)
+            del self._embedding_
             return data
         return None  # is this even possible?
 
     def __getstate__(self):
         state = self.__dict__.copy()
 
-        if "embedding_" in state:
-            state["embedding_"] = cudf.DataFrame.from_gpu_matrix(
-                state["embedding_"])
+        if "_embedding_" in state:
+            state["_embedding_"] = cudf.DataFrame.from_gpu_matrix(
+                state["_embedding_"].to_output('cupy'))
```

Comment:
this shouldn't be needed anymore, CumlArray is picklable

Reply:
I cleaned it up. Thank you.